c7n/resources/ec2.py

```diff
@@ -197,7 +197,13 @@ def get_resource_date(self, i):
         v = i.get('StateTransitionReason')
         if not v:
             return None
-        return parse(self.RE_PARSE_AGE.findall(v)[0][1:-1])
+        dates = self.RE_PARSE_AGE.findall(v)
+        if dates:
+            try:
+                return parse(dates[0][1:-1])
+            except ValueError:
+                self.log.warning('Failed to parse date in EC2 state-age ({})'.format(v))
```

Comment:
i'd prefer this continue to raise if its given an invalid date, the known case where this isn't a date is handled by the conditional.